src/Login.ts

```diff
@@ -123,8 +124,11 @@ export default class Login {
 
     public async getFlows(): Promise<Array<LoginFlow>> {
         const client = this.createTemporaryClient();
-        const { flows } = await client.loginFlows();
-        this.flows = flows;
+        const { flows }: { flows: LoginFlow[] } = await client.loginFlows();
+        // If an m.login.sso flow is present that is flagged as being for MSC3824 OIDC compatibility then we only return that flow
+        const oidcCompatibilityFlow =
+            flows.find(f => f.type === "m.login.sso" && f["org.matrix.msc3824.delegated_oidc_compatibility"]);
+        this.flows = oidcCompatibilityFlow ? [oidcCompatibilityFlow] : flows;
```

Comment:
What if we used `filter` instead of `find` and then just did `this.flow = oidcCompatibilityFlows || flows`?

Reply:
`[] || flows` doesn't work as providing a default (becuase `[]` is truthy). Also, the goal (as per comment) is to only return a single (the first) compatibility flow if one is present.